src/index.ts

```diff
@@ -18,6 +18,8 @@ export interface JsonFetchOptions extends Omit<RequestInit, 'body'> {
   retry?: Parameters<typeof promiseRetry>[0];
   timeout?: number;
   expectedStatuses?: Array<number>;
+  onRequestStart?: (url: string, retryCount: number) => void;
```

Comment:
Maybe make this an object, so we can add new attributes to it in the future without having to worry about order.